.ci/test_r_package_windows.ps1

```diff
@@ -283,7 +283,7 @@ if ($env:R_BUILD_TYPE -eq "cmake") {
 if ($env:COMPILER -eq "MSVC") {
   Write-Output "Running tests with testthat.R"
   cd R-package/tests
-  Run-R-Code-Redirect-Stderr "source('testthat.R')" ; Check-Output $?
+  Rscript.exe --vanilla "testthat.R" ; Check-Output $?
```

Comment:
Please comment why you use `Rscript.exe` here to prevent accidental reversion of this PR in the future.

Reply:
Added a comment in https://github.com/microsoft/LightGBM/pull/5448/commits/121e117edeb734e320cc85f1bd0a6660557823ef.